### PR TITLE
Refactor Segmentation models

### LINKED ISSUE #4676: Refactor Models builders to make them reusable on Prototype
### 🚀 The feature

The current model builders are hard to reuse from the prototype area, leading to large amounts of copy-pasting. It would be good to refactor the space in a BC manner so that we can reuse as many parts as we can:
- [x] Segmentation models:
  - Update them to follow the standard convention of defining the builders on the same file as their class.
  - Make private model builders reusable from prototype.
  - Clean up the implementations and expose publicly variables that are currently pushed to kwargs.
- [x] Detection models:
  - Make private model builders reusable from prototype.
  - Refactor the VGG16 Features weights mechanism on SSD to work well with the proposed weights prototype. 

### Motivation, pitch

-

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by datumbox**
Fixes #4676

This PR refactors the entire `segmentation` package to align with the conventions followed by `classification`, `detection` and `quantization`. 

Here are the details:
- The public model builders located at `models.segmentation.segmentation` now move to their respective model packages. For example, `fcn_resnet50()` is now defined in `fcn.py`.
- The private builder methods of each Model are refactored, so that they can be reused on `prototype` with minimal copy-pasting.
- We expose publicly `pretrained_backbone` instead via `kwargs`. 

Accuracy remains the same for all pre-trained models. No modifications on tests required because the changes are fully-BC. This PR enables the work on #4611.

### CODE REVIEW

**Comment by datumbox** [comment]
Thanks for the reviews. I confirmed that all models maintain the right model validation statistics when compared to the main branch:
```
torchrun --nproc_per_node=2 train.py --test-only --pretrained --model fcn_resnet50
global correct: 91.4
mean IoU: 60.5

torchrun --nproc_per_node=2 train.py --test-only --pretrained --model fcn_resnet101
global correct: 91.9
mean IoU: 63.7

torchrun --nproc_per_node=2 train.py --test-only --pretrained --model deeplabv3_resnet50
global correct: 92.4
mean IoU: 66.4

torchrun --nproc_per_node=2 train.py --test-only --pretrained --model deeplabv3_resnet101
global correct: 92.4
mean IoU: 67.4

torchrun --nproc_per_node=2 train.py --test-only --pretrained --model deeplabv3_mobilenet_v3_large
global correct: 91.2
mean IoU: 60.3

torchrun --nproc_per_node=2 train.py --test-only --pretrained --model lraspp_mobilenet_v3_large
global correct: 91.2
mean IoU: 57.9
```

I'll merge once the CI passes.
